maintaining-and-tuning/monitoring.md

```diff
@@ -198,16 +199,189 @@ These instructions assume that GRR server and Prometheus are both up and running
+panels; each such
+[panel](https://grafana.com/docs/grafana/latest/panels/panels-overview/#panel-overview)
+consists of a graph that contain one or more metrics queried from Prometheus,
+and possibly alerting rules.
+
+1. If you want to use the alerts, you first need to define a
+[notification channel](https://grafana.com/docs/grafana/latest/alerting/notifications/#alert-notifications)
+for the alerts. This can be done by heading over to
+`http://localhost:3000/alerting/notification/new` and
+[following the form](https://grafana.com/docs/grafana/latest/alerting/notifications/#new-notification-channel-fields)
+to add a notification channel. Once a notification channel is set up,
+you will start receiving alerts from the existing dashboards, as those
+contain definitions for simple alerts. There are two sample alerting
+rules: in each dashboard, there is a panel that counts the number of
+active processes for the specific job. For example, in the Workers
+dashboards, there is a panel called "Active Processes" which shows
+the current number of active Workers processes. If the number of active
+workers is zero - an alert will be fired. If you want to disable or
+remove an alert, go to the dashboard and the corresponding panel,
+and there you can remove the alerting rule.
+
+1. You can now use the dashboards. The dashboards can give a general
+overview over the main components of the GRR server and Fleetspeak
+servers, which can be utilized by the user to monitor different metrics
+of each component. Examples for such metrics can be found in the
+[examples above](#example-queries).
+Remember that the dashboards and alerts are flexible, and can be
+expanded or modified to adjust to your exact needs. Additional metrics
+can be used by exploring `http://<host>:<port>/metrics` for each
+component of GRR server (change the port according to the GRR server
+component you want) and Fleetspeak servers, and if you'd like to
+create your own
+[custom dashboards](https://grafana.com/docs/grafana/latest/getting-started/getting-started/#create-a-dashboard),
+[panels](https://grafana.com/docs/grafana/latest/panels/add-a-panel/#add-a-panel)
+and [alerts](https://grafana.com/docs/grafana/latest/alerting/create-alerts/#create-alerts),
+make sure to go over the corresponding documentation in Grafana.
 
-1. [Set up Prometheus as a data source](https://grafana.com/docs/grafana/latest/features/datasources/prometheus/#prometheus-data-source) for Grafana, so that Grafana can display the metrics from Prometheus. To do that, follow the guide [here](https://grafana.com/docs/grafana/latest/features/datasources/add-a-data-source/#add-a-data-source) (in step 3 of the guide, we suggest naming the data source `grr-server` so that it will match the configuration file for the provided sample dashboards later on. If it is a Fleetspeak-enabled GRR deployment, add a second datasource named `fleetspeak`).
+## Real-World Setup
```

Comment:
I'm not sure what a "Real World Setup" header means. What is this section about?

Reply:
This section provides brief information on service discovery in Prometheus and the security considerations. Do you think it should be elaborated or removed?